collaboration-engine/src/main/java/com/vaadin/collaborationengine/PropertyChangeHandler.java

```diff
@@ -43,6 +45,36 @@ interface PropertyChangeEvent {
          * @return the value, not {@code null}
          */
         Object getValue();
+
+        /**
+         * Gets the new value as an instance of the given class.
+         *
+         * @param type
+         *            the expected type of the returned instance
+         * @param <T>
+         *            the type of the value from <code>type</code> parameter,
+         *            e.g. <code>String</code>
+         * @return throws an {@link UnsupportedOperationException}
+         */
+        default <T> T getValue(Class<T> type) {
+            throw new UnsupportedOperationException(
+                    "This method is not implemented");
+        }
+
+        /**
+         * Gets the new value as an instance of the given type reference.
+         *
+         * @param typeRef
+         *            the expected type reference of the returned instance
+         * @param <T>
+         *            the type reference of the value from <code>typeRef</code>
+         *            parameter, e.g. <code>List<String>></code>
+         * @return throws an {@link UnsupportedOperationException}
+         */
+        default <T> T getValue(TypeReference<T> typeRef) {
+            throw new UnsupportedOperationException(
+                    "This method is not implemented");
```

Comment:
I couldn't find implementing classes of this interface in the Collaboration Kit repository other than the `DefaultPropertyChangeEvent` in `FormManager`. Do users of Collaboration Kit have their own implementations of `PropertyChangeEvent`? If this is just to add the API to `FormManager` to specify class or type reference, then these methods do not need to be declared with a `default` implementation.

I get that this was done so that it doesn't affect current code, but if there isn't any code to break we might be able to get away with not providing a `default` implementation. I doubt users of Collaboration Kit have their own implementation of `PropertyChangeEvent`.

Reply:
The interface is public, so part of the API. If a developer creates their own class based on `AbstractCollaborationManager` they need to implement this interface internally.

I agree this is fairly unlikely that someone has done this, but it's at least possible,